### PR TITLE
feat(frontend): adjusted errors handling for swap in service and component

### PR DESCRIPTION
**Pull request by roman-nazaruk**
# Motivation

Adjusted Swap service to throw and handle errors in the Swap component

# Changes
- Refactored Swap service to throw errors using throwSwapError
- Updated error handling logic in the Swap component
- Added new properties to Plausible tracking events

### CODE REVIEW

File: src/frontend/src/lib/components/swap/SwapWizard.svelte [hunk 1]
```diff
@@ -113,7 +114,7 @@
 				}
 			});
 
-			setTimeout(() => close(), 750);
+			setTimeout(() => close(), 50);
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/swap/SwapWizard.svelte]: why this change?

File: src/frontend/src/lib/components/swap/SwapWizard.svelte [hunk 2]
```diff
@@ -127,23 +128,26 @@
...
-						url: `https://app.icpswap.com/swap?input=${$sourceToken.ledgerCanisterId}&output=${$destinationToken.ledgerCanisterId}`,
-						text: 'icpswap.com'
-					}
-				});
-			} else if (errorDetail && errorDetail.startsWith('Swap failed.')) {
-				failedSwapError.set({ message: errorDetail, variant: 'error' });
+			}
+
+			if (err instanceof SwapError) {
+				if (err.code === 'withdraw_failed') {
```
Comment (AntonioVentilii) [inline src/frontend/src/lib/components/swap/SwapWizard.svelte]: can we have enums for the codes?

**Review by AntonioVentilii — COMMENTED** [review]
Some tests? Even for hedge cases

File: src/frontend/src/lib/components/swap/SwapWizard.svelte (hunk 1 above)
Comment (roman-nazaruk) [reply src/frontend/src/lib/components/swap/SwapWizard.svelte]: I added this change because we had a long loading spinner. Since we call triggerWallet right after a successful swap, it was causing two spinners to appear briefly.

**Comment by roman-nazaruk** [comment]
> Some tests? Even for hedge cases

TBH, it's almost impossible, since the swap method is called inside a child component and isn't exported. I have no clear way to mock or test it.

File: src/frontend/src/lib/components/swap/SwapWizard.svelte (hunk 1 above)
Comment (AntonioVentilii) [reply src/frontend/src/lib/components/swap/SwapWizard.svelte]: then, maybe just `debounce` the call? does it work?

File: src/frontend/src/tests/lib/utils/swap.utils.spec.ts [hunk 3]
```diff
@@ -450,4 +452,25 @@ describe('swap utils', () => {
...
+
+			expect(isSwapError(error)).toBeFalsy();
+		});
+
+		it('should return false for non-error values', () => {
+			expect(isSwapError(null)).toBeFalsy();
+			expect(isSwapError(undefined)).toBeFalsy();
+			expect(isSwapError('string')).toBeFalsy();
+			expect(isSwapError({ code: 'deposit_error' })).toBeFalsy();
+		});
```
Comment (AntonioVentilii) [inline src/frontend/src/tests/lib/utils/swap.utils.spec.ts:474]: nice!!!

File: src/frontend/src/lib/components/swap/SwapWizard.svelte (hunk 1 above)
Comment (roman-nazaruk) [reply src/frontend/src/lib/components/swap/SwapWizard.svelte]: Nice catch, I took a dipper look on this and fix was not related to this.
